packages/frontend/src/components/helpers/PseudoListItem.tsx

```diff
@@ -177,19 +176,35 @@ export const PseudoListItemAddContactOrGroupFromInviteLink = ({
     }
   }, [accountId, parsedQr])
 
-  return (
+  const onClick = () => processQr(accountId, inviteLinkTrimmed)
+
+  return contact ? (
+    <ContactListItem
+      showCheckbox={false}
+      checked={false}
+      showRemove={false}
+      contact={contact}
+      onClick={onClick}
+    />
+  ) : parsedQr?.kind === 'askVerifyGroup' ? (
     <PseudoListItem
-      id='newcontactorgroupfrominvitelink'
+      id='newgroupfrominvitelink'
       cutoff='+'
-      text={
-        parsedQr?.kind === 'askVerifyGroup'
-          ? tx('menu_new_group')
-          : tx('menu_new_contact')
-      }
-      subText={
-        parsedQr?.kind === 'askVerifyGroup' ? parsedQr.grpname : contactName
-      }
-      onClick={() => processQr(accountId, inviteLinkTrimmed)}
+      text={parsedQr.grpname}
+      subText={tx('join_group')}
+      onClick={onClick}
+    />
+  ) : (
+    // There are many kinds of QRs. For example, if you paste an invite link
+    // of a group that you created yourself, you'll get
+    // `parsedQr.kind === "withdrawVerifyGroup"`.
+    // TODO we probably want to better handle such cases.
```

Comment:
do we want to handle all qrlinks in search fields now?

Reply:
You mean in this MR? I think not. I think we covered the two most common cases. The rest can be improved later. Anyways it's just a matter of proper labeling. The dialog that opens when you click the chatlist item will tell the user more.

Reply:
might not be a bad idea to handle other qr code links as well there, as this is where some users seem to paste it to "scan" it. But thats for a future PR, I just asked because of the comment.